app/models/user.rb

```diff
@@ -9,6 +9,11 @@ class User < ApplicationRecord
   has_many :cars
 
   validates :name, presence: true
+  before_save :set_regular_user
```

Comment:
In generally we should avoid 'Callbacks'. Callbacks are bad :)
https://medium.com/planet-arkency/the-biggest-rails-code-smell-you-should-avoid-to-keep-your-app-healthy-a61fd75ab2d3

Except that, your solution doesn't work correct now. Please make model Tests for User (3 scenarios)

1) save User without 'type' value. Should get 'RegularUser')
2) save User with 'type' value 'RegularUser'. Should get 'RegularUser'
3) save User with 'type' value 'AdminUser'. Should get 'AdminUser'

Last case in your solution doesn't work. In fact you can't add an AdminUser usingn AdminUser.create!...

Tip: You don't need use 'Devise' to test User. Look at seeds, what information you need to add a User. Make a User factory with this mention information and use it.

Finally as suggested add please a migration which sets default type to 'RegularUser'. This is a neat and very simple solution. Why didn't you use it?